### PR TITLE
chore: readmes, some docs, ci and tests

### PR DESCRIPTION
**Pull request by perfectmak**
This sets up a couple of things to make it easier to understands the codebase and also help with future improvements,refactoring and debugging.

Some of the things added in this PR are:
- Add some readmes with basic getting started instructions
- Setup jest for running tests (and included some preliminary tests for graphql utils)
- Setup a github actions workflow to run build and tests.
- Also, added some docs to explain of the graphql utils behaviour.

### CODE REVIEW

File: test/unit/checkpoint/graphql/__snapshots__/utils.test.ts.snap [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    orderDirection: String
+    where: WhereVote
+  ): [Vote]
+  vote(id: String): Vote
+}
+
+
+input WhereVote {
+  id: null
+  id_in: [null]
```
Comment (perfectmak) [inline test/unit/checkpoint/graphql/__snapshots__/utils.test.ts.snap:18]: this looks off, the types generated are `null`, I'm assuming they should be `Int` based on the `type Vote` defined. @bonustrack is this the expected behavior or it is a bug we should look into?

Comment (bonustrack) [reply test/unit/checkpoint/graphql/__snapshots__/utils.test.ts.snap:18]: That's right, based on the schema defined it should return a type `Int`. I think the setup I've done assume the id is a String see https://github.com/snapshot-labs/sx-api/blob/master/src/checkpoint/graphql/utils.ts#L86 but there is no reason to enforce that, a Int should work as id too.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# SX-API
+
+This API uses checkpoint to index Snapshot X spaces information on Starknet.
```
Comment (bonustrack) [inline README.md:3]:
```suggestion
This API uses Checkpoint to index Snapshot X spaces information on StarkNet.
```

**Review by bonustrack — APPROVED** [review]
Awesome, tested locally and it works well. Just a detail but I would recommend when writing readme or comments to use uppercase on first letter for proper nouns like "Checkpoint" instead of "checkpoint" to not be confused (unless you talking about actual checkpoints and not the solution), also use StarkNet (with the official casing) and not starknet or Starknet.

**Pull request merged**